src/collection/historical_signatures/signature_collector.py

```diff
@@ -0,0 +1,215 @@
+"""
+Module contains `SignatureCollector`, class dedicated to collection of history of transaction on Solana chain for
+lending protocols.
+Collection logic:
+    t_O - watershed block, block that we assign as divider of historical and current data.
+    Takes `t_0` from db.
+    PPK (protocol public key) is collected from environmental variable.
+    Collects all historical data as follows:
+    1) starts from any transaction (`tx_0`) from block `t_0`.
+    2) Collects 1000 tx signatures starting with `tx_0`.
+    3) Collected transactions belong to sequence of blocks from `t_0` to `t_0 - k`.
+    Store to tx_signatures table all transactions from blocks `t_0 - 1` to `t_0 - k + 1` with flag `from signatures`.
+    If signature already exists - change flag to `from_signatures`.
+    Transactions from block `t_0` are handled in current data collector.
+    4) we get presumably the earliest transaction signature from `t_0 - k + 1` block and collect next 1000 tx.
+    5) repeat 3-4 until reaching the first transaction.
+    6) As first (earliest) transaction is reached - consider historical data (signatures only) for given PPK
+    at time t_0 collected.
+    If being restarted - fetch block number from last tx signature for given PPK and flag `signature`.
+    Use this block instead of `t_0`
+"""
+import logging
+import os
+import time
+import traceback
+from typing import List
+
+from solders.rpc.responses import RpcConfirmedTransactionStatusWithSignature
+from solders.signature import Signature
+from solders.transaction_status import TransactionErrorFieldless
+from solana.exceptions import SolanaRpcException
+import solana.rpc.api
+
+from src.collection.shared.generic_collector import GenericSolanaConnector, log_performance_time
+import db
+
+
+LOGGER = logging.getLogger(__name__)
+
+TX_BATCH_SIZE = 1000
+
+
+class SignatureCollector(GenericSolanaConnector):
+    def __init__(self):
+        super().__init__()
+        self.protocol = os.getenv("PROTOCOL_PUBLIC_KEY", "")
+        self._oldest_signature: Signature | None = None  # The oldest signature from the oldest completed block
+        self._signatures_completed: bool = False
+        self._oldest_completed_slot: int = 0
```

Comment:
This default value is unfortunate because if the oldest completed slot is indeed 0, then in such a scenario, we already have all the signatures we need.

Reply:
Changed to -1. I meant to use any unrealistic number, and even thou zero is perfectly unrealistic - it also might be confusing.